File: TowerDefence.spritebuilder/Unit/Tower/TowerFabric.swift
//
//  TowerFabric.swift
//  TowerDefence
//
//  Created by koveres on 08.12.14.
//  Copyright (c) 2014 Apportable. All rights reserved.
//

import Foundation


/**
*
* @author vereshchagin
*/
public class TowerFabric : StackObjectAbstractFabric{
    
    
    public  override init(intlist:[CInt] )
    {
       
        super.init(intlist: intlist);
        
    }
    
    public func getTower(  idTower:TowerType,  position1:CGPoint, solderFinder1:TargetFinder)   -> (Tower)
    {
        var newobj :Tower ;
        position = position1;
        solderFinder = solderFinder1;
        newobj =    getObject( idTower.rawValue ) as Tower;
        return newobj;
    
    }
    
    var position:CGPoint?;
    var solderFinder:TargetFinder?;
 
    override func getNewObject( idObject:CInt, grObj:AnyObject) ->(StackObject) {
        var idTower:TowerType? = TowerType(rawValue: idObject);
        if(idTower != nil){
        switch idTower!
        {
        case TowerType.Tower1:
            return   Tower1(position: position!,solderFinder: solderFinder!,grObj: grObj);
        case TowerType.Tower2:
            return   Tower2(position: position!,solderFinder: solderFinder!,grObj: grObj);
        case TowerType.Tower3:
            return   Tower3(position: position!,solderFinder: solderFinder!,grObj: grObj);
            }
        }
        return Tower1(position: position!,solderFinder: solderFinder!,grObj: grObj);
    }
}

File: test.spritebuilder/Source/TDEngine/Unit/Tower/TowerFabric.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package Unit.Tower;

import Utils.Point2D;
import Utils.StackObject;
import Utils.StackObject.StackObjectAlreadyEngaged;
import Utils.StackObjectAbstractFabric;
import internalObject.TowerInternal;
import internalObject.TowerInternal.TowerType;
import java.util.List;
 

/**
 *
 * @author vereshchagin
 */
public class TowerFabric extends StackObjectAbstractFabric{

    public TowerFabric(List<Integer> intlist)
    {
        super(intlist); 
    }
    
    public Tower getTower(TowerInternal.TowerType idTower,Point2D position,TargetFinder solderFinder) throws StackObjectAlreadyEngaged 
    {
        Tower newobj;
        this.position = position;
        this.solderFinder = solderFinder;
        newobj = (Tower) this.getObject( idTower.ordinal() );
        newobj.init(position,solderFinder);
        return newobj;
        
         
    }
        Point2D position;
        TargetFinder solderFinder;
    @Override
    protected StackObject getNewObject(Integer idObject) {
         TowerType idTower;
       idTower = TowerType.values()[idObject]; 
       switch(idTower)
        {
            case Tower1:
                return new Tower1(position,solderFinder);
            case Tower2:
                return new Tower2(position,solderFinder);
            case Tower3:
                return new Tower3(position,solderFinder);
        }
        return new Tower1(position,solderFinder);
    }
}
